Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
-            cell.setup(transaction.inputs, presentingController: self)
-            cell.expandOrCollapse = { [weak self] in
-                self?.tableTransactionDetails.reloadData()
+            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionDetailCell") as! TransactionDetailCell
+            cell.txDetail = self.generalTxDetails[indexPath.row]
+            cell.onTxDetailValueCopied = { copiedDetail in
+                Utils.showBanner(in: self.view.subviews.first!, type: .success, text: String(format: LocalizedStrings.sgCopied, copiedDetail))
             }
             return cell
-            
+
         case 2:
-            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactiontOutputDetailsCell") as! TransactiontOutputDetailsCell
-            cell.setup(transaction.outputs, presentingController: self)
-            cell.expandOrCollapse = { [weak self] in
-                self?.tableTransactionDetails.reloadData()
+            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionInputDetailCell") as! TransactionInputDetailCell
+            cell.setup(transaction.inputs[indexPath.row])
+            cell.onTxHashCopied = {
+                Utils.showBanner(in: self.view.subviews.first!, type: .success, text: LocalizedStrings.previousOutpointCopied)
+            }
+            return cell
+
+        case 3:
+            let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionOutputDetailCell") as! TransactionOutputDetailCell
+            cell.setup(transaction.outputs[indexPath.row])
+            cell.onTxHashCopied = {
+                Utils.showBanner(in: self.view.subviews.first!, type: .success, text: LocalizedStrings.addrCopied)
             }
             return cell
-            
+
+        case 4:
+            return tableView.dequeueReusableCell(withIdentifier: "TransactionViewOnDcrdataCell")!
+
         default:
             return UITableViewCell()
         }
     }
-    
+
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if indexPath.section == 0 && indexPath.row == 6 {
-            self.copyText(self.transaction.hash)
+        if indexPath.section == 4 {
+            if BuildConfig.IsTestNet {
+                self.openLink(urlString: "https://testnet.dcrdata.org/tx/\(self.transaction.hash)")
+             } else {
+                self.openLink(urlString: "https://dcrdata.decred.org/tx/\(self.transaction.hash)")
+            }
+        }
+    }
+
+    func openLink(urlString: String) {
+        //TODO
+        if let url = URL(string: urlString) {
+            UIApplication.shared.open(url)
+    //            let viewController = SFSafariViewController(url: url)
+    //            viewController.delegate = self as SFSafariViewControllerDelegate
+    //            self.navigationController?.pushViewController(viewController, animated: true)
```

Comment:
What's wrong with this code? If it does not work as we would like or expect it to work, let's delete it and also delete the `SafariServices` import.